### PR TITLE
Adding Gazebo Testing Procedure to README

### PR DESCRIPTION
**Pull request by garethellis0**

### CODE REVIEW

**Comment by garethellis0** [comment]
@ValRat I'd like to get some feedback on this, let me know if you think it's confusing at all, or if there's anything I should add

**Comment by garethellis0** [comment]
Welp. If the intent is the for the testers to easily be able to add failing tests, then having them add internal unit-tests might be a little hard. Rostest, which allows you to test nodes by sending in data and examinging the outputs, might be a better choice. That being said, will have to add a new section to the readme for that.

File: README.md [hunk 1]
```diff
@@ -121,9 +125,13 @@ some_ros_package
...
 - To manually control the robot, run `rosrun turtlesim turtle_teleop_key /turtle1/cmd_vel:=/robot/cmd_vel`
 
-## Github Procedure
-- We follow the "Feature Branch Workflow"
-- A good tutorial can be found [here](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow)
+## Gazebo Testing Procedure
+1. Ask the node creator for a generally specification as to how the node should behave
+- Try and come up with awkward situations which could break the node or cause it to misbehave
+- Test the specification provided by the original node creator, plus any awkward situations you came up with, in Gazebo
+- If the node does not behave as expected (for example, produces the incorrect output) then add unit test(s) to the node (there should already be a testing file present) that mimics the situation(s) that failed in Gazebo (so they should fail if you run the unit tests). **All** Unit Tests should be on a new branch named `NODE-NAME-gazebo-testing`
```
Comment (garethellis0) [inline README.md]: Should probably be a rostest, rather then a standard gtest unit test, waiting on #66, #67

Comment (ValRat) [inline README.md]: Should we consider creating a rosbag to store what caused the crash and replay it for the rostest?

**Comment by garethellis0** [comment]
This is such a low priority right now, and I'm cleaning out PR's

**Comment by ValRat** [comment]
We can return to this if we need be, but deleting branch for cleanup.

**Pull request closed without merging**